src/api/app/services/notifications_fetcher.rb

```diff
@@ -0,0 +1,25 @@
+class NotificationsFetcher
+  attr_reader :notifications
+
+  def initialize(scope:, type: :requests, user: User.session, by_project: nil)
+    @scope = scope
+    @user = user
+    @type = (type || :requests).to_sym
+    @by_project = by_project
+  end
+
+  def valid?
+    return false unless [:requests, :comments, :incoming_requests, :outgoing_requests].include?(@type)
```

Comment:
This method could be a one-liner. I would also rename it to `valid_type?`.

```ruby
def valid_type?
  [:requests, :comments, :incoming_requests, :outgoing_requests].include?(@type)
end
```

Reply:
yes definitely, just that I'm still not sure if we will need other validation. In between we were validating as well the state, i.e (read/unread) but now we are passing it in the scope... I will keep it in mind. Either way, the method will still be called `valid?` because it's a common pattern that we use all around our services.